code/datums/mapgen/dungeon_generators/maintenance_generator/maintenance_room_themes/random.dm

```diff
@@ -261,3 +264,213 @@
+		/obj/item/stack/spacecash/c1000 = 2,
+		/obj/item/stack/sheet/mineral/diamond = 5
+		)
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/hivebot/range = 2,
+		)
+
+///mama mia
+/datum/dungeon_room_theme/maintenance/kitchen
+	weighted_feature_spawn_list = list(
+		/obj/machinery/griddle = 1,
+		/obj/machinery/microwave = 1,
+		/obj/structure/closet/secure_closet/freezer/fridge = 1,
+		/obj/item/storage/box/donkpockets = 1,
+		/obj/item/kitchen/knife = 1,
+		/obj/effect/spawner/lootdrop/random_meat = 5
+
+		)
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/retaliate/goat = 2
+		)
+
+///just lik the founding fathers intended
+/datum/dungeon_room_theme/maintenance/ancient_armory
+	weighted_possible_floor_types = list(
+		/turf/open/floor/plasteel/dark = 3,
+		/turf/open/floor/plasteel = 5,
+		/turf/open/floor/plating = 3, 
+		)
+	
+	weighted_feature_spawn_list = list(
+		/obj/structure/filingcabinet/chestdrawer = 1,
+		/obj/item/kirbyplants/random  = 2,
+		list(/obj/structure/rack , /obj/item/melee/spear/plugged_musket) = 1,
+		list(/obj/structure/rack , /obj/item/clothing/suit/armor/vest, /obj/item/clothing/head/helmet/riot) = 1,
+		/obj/structure/frame/machine = 1,
+		/obj/structure/frame/computer = 1,
+		/obj/effect/mine/stun	
+		)
+	
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		)
+/datum/dungeon_room_theme/maintenance/ancient_armory/pre_initialize()
+
+	. = ..()
+	for(var/i in 1 to 3)
+		if(prob(10))
+			weighted_feature_spawn_list[/obj/item/kitchen/knife/combat/bone]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
+
+///these don't actually spawn inside the suit storage and I think thats funny
+/datum/dungeon_room_theme/maintenance/hardsuit
+	weighted_feature_spawn_list = list(
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/mining),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/ancient),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/engine),
+		list(/obj/machinery/suit_storage_unit,/obj/item/clothing/suit/space/hardsuit/medical)
+		)
+
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		/mob/living/simple_animal/hostile/robot/burst = 2
+		)
+
+///this gun blows so much ass using it is actually more likely to get you killed
+/datum/dungeon_room_theme/maintenance/lasgun
+
+	weighted_mob_spawn_list = list(
+		/mob/living/simple_animal/hostile/robot/burst = 2,
+		/mob/living/simple_animal/hostile/robot/advanced/ranged = 1
+		)
+
+/datum/dungeon_room_theme/maintenance/lasgun/pre_initialize()
+	. = ..()
+	for(var/i in 1 to 5)
+		if(prob(10))
+			weighted_feature_spawn_list[/obj/item/gun/energy/laser/captain/scattershot ]++
+		else
+			weighted_feature_spawn_list[/obj/item/melee/spear/bonespear/chitinspear]++
```

Comment:
just remove this entirely if it's just gonna spawn the spear like the ancient armory

Reply:
bye lasgun